### ISSUE TITLE
[Feature Request] Tree component.

### ISSUE THREAD

**Issue by rainjay** [mantinedev/mantine#756]
### Usage example, including component/hook API

https://github.com/react-component/tree
https://github.com/react-component/tree-select

**Comment by zhulien-ivanov** [comment]
Those are the only 2 essential components for a UI library that are still missing. We're planning migration to Mantine but can't do it without a TreeSelect of some sort. Do you have any plans on implementing this for the next major release?

**Comment by rtivital** [comment]
Yes, I'm planning to add these components in one of 4.x releases

**Comment by pavi2410** [comment]
@rtivital Let me know if you require any help. I need this component.

**Comment by rtivital** [comment]
You can contribute a TreeSelect component, to do that:
- target `dev` branch
- add component to `@mantine/labs` package
- if you have any questions, contact on Discord in Contributing channel

**Comment by stephenirven** [comment]
I'm also interested in this. For info of anyone looking for a solution right now, it could be worth looking at this [https://www.npmjs.com/package/@minoru/react-dnd-treeview](https://www.npmjs.com/package/@minoru/react-dnd-treeview)

It seems to be quite fully featured, and there are CodePen samples for various use cases including select. The samples use Material-UI, but the module doesn't depend on it, so it's possible to get it working with Mantine.
